### PR TITLE
Feature/ios GitHub actions

### PR DESCRIPTION
**Pull request by mateki0**

### CODE REVIEW

File: .github/workflows/apple_app_store_publish.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      - uses: actions/checkout@v3
+      - name: Prepare env file
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16.14
```
Comment (jan-kozinski) [inline .github/workflows/apple_app_store_publish.yml:28]:
```suggestion
- name: Setup Node.js
  uses: actions/setup-node@v3
  with:
    node-version: '16.14'
    cache: 'yarn'
```

File: .github/workflows/apple_app_store_publish.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+      - name: Restore node_modules from cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
```
Comment (jan-kozinski) [inline .github/workflows/apple_app_store_publish.yml]:
```suggestion
- name: Setup Node.js
  uses: actions/setup-node@v3
  with:
    node-version: '16.14'
    cache: 'yarn'
```

File: .github/workflows/apple_app_store_publish.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+      - name: Bundle and Upload to App Store
+        env:
+          KEY_ID: ${{secrets.APPLE_KEY_ID}}
+          ISSUER_ID: ${{secrets.APPLE_ISSUER_ID}}
+          IOS_AUTH_KEY: ${{secrets.IOS_AUTH_KEY_BASE64}}
+          MATCH_PASSWORD: ${{secrets.MATCH_PASSWORD}}
+          MATCH_GIT_BASIC_AUTHORIZATION: ${{secrets.MATCH_GIT_BASIC_AUTHORIZATION}}
+          PAT: ${{secrets.PAT}}
+        run: bundle exec fastlane build_and_release_to_app_store
```
Comment (jan-kozinski) [inline .github/workflows/apple_app_store_publish.yml:55]: we might want to commit the app version bump to the repo after publishing

File: .github/workflows/apple_testflight_publish.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+          node-version: 16.14
+
+      - name: Restore node_modules from cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
```
Comment (jan-kozinski) [inline .github/workflows/apple_testflight_publish.yml]:
```suggestion
- name: Setup Node.js
  uses: actions/setup-node@v3
  with:
    node-version: '16.14'
    cache: 'yarn'
```

File: .github/workflows/apple_app_store_publish.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  release-ios-production:
+    name: PRODUCTION | Build and release iOS app
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Prepare env file
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.14
```
Comment (jan-kozinski) [inline .github/workflows/apple_app_store_publish.yml:24]:
```suggestion
          node-version: 16.14
          cache: yarn
```

**Review by jan-kozinski — CHANGES_REQUESTED** [review]

File: .github/workflows/apple_testflight_publish.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  release-ios-testflight:
+    name: TESTFLIGHT | Build and release iOS app
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Prepare env file
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.14
```
Comment (jan-kozinski) [inline .github/workflows/apple_testflight_publish.yml:24]:
```suggestion
          node-version: 16.14
          cache: yarn
```

File: .github/workflows/apple_testflight_publish.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
```
Comment (mateki0) [inline .github/workflows/apple_testflight_publish.yml]:
```suggestion
      - uses: actions/setup-node@v3
        with:
          node-version: 16.14
          cache: yarn

```

**Review by jan-kozinski — APPROVED** [review]

**Pull request merged**